src/btreemap.rs

```diff
@@ -353,6 +365,18 @@ where
     ///   key.to_bytes().len() <= max_size(Key)
     ///   value.to_bytes().len() <= max_size(Value)
     pub fn insert(&mut self, key: K, value: V) -> Option<V> {
+        if let Some((first_key, _)) = &self.first_key_value {
```

Comment:
Since both first and last entry caches are touched in this method I suggest you create a single `first_last_entry_cache` wrapped by `RefCell` so that it calls `borrow_mut()` only once and then decide which part to update. inside you can store a pair and handle a special case when BTM.len()==1 to always use one of the cells.